mesheryctl/internal/cli/root/perf/perf.go

```diff
@@ -129,79 +127,75 @@ var PerfCmd = &cobra.Command{
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
 		// Importing SMPS Configuration from the file
+		var req *http.Request
 		if filePath != "" {
-			var t models.PerformanceSpec
-			err := yaml.Unmarshal([]byte(filePath), &t)
+			smpsConfig, err := ioutil.ReadFile(filePath)
+
 			if err != nil {
-				return errors.Wrapf(err, utils.PerfError(fmt.Sprintf("failed to unmarshal yaml file %s", filePath)))
-			}
-			if testDuration == "" {
-				testDuration = fmt.Sprintf("%fs", t.EndTime.Sub(t.StartTime).Seconds())
+				return err
 			}
-			if testURL == "" {
-				testURL = t.EndpointURL
-			}
-			if concurrentRequests == "" {
-				concurrentRequests = fmt.Sprintf("%d", t.Client.Connections)
+
+			req, err = http.NewRequest("POST", mctlCfg.GetPerf().LoadTestSmpsURI, bytes.NewBuffer(smpsConfig))
+			if err != nil {
+				return errors.Wrapf(err, utils.PerfError(fmt.Sprintf("failed to create new request to %s", mctlCfg.GetBaseMesheryURL())))
 			}
-			if qps == "" {
-				qps = fmt.Sprintf("%f", t.Client.Rps)
+		} else {
+
+			if testName == "" {
+				log.Debug("Test Name not provided")
+				testName = StringWithCharset(8)
+				log.Debug("Using random test name: ", testName)
 			}
-		}
 
-		if testName == "" {
-			log.Debug("Test Name not provided")
-			testName = StringWithCharset(8)
-			log.Debug("Using random test name: ", testName)
-		}
+			postData := ""
 
-		postData := ""
+			startTime := time.Now()
+			duration, err := time.ParseDuration(testDuration)
+			if err != nil {
+				return errors.Wrapf(err, utils.PerfError(fmt.Sprintf("failed to parse test duration %s", testDuration)))
+			}
 
-		startTime := time.Now()
-		duration, err := time.ParseDuration(testDuration)
-		if err != nil {
-			return errors.Wrapf(err, utils.PerfError(fmt.Sprintf("failed to parse test duration %s", testDuration)))
-		}
+			endTime := startTime.Add(duration)
 
-		endTime := startTime.Add(duration)
+			postData = postData + "start_time: " + startTime.Format(time.RFC3339)
+			postData = postData + "\nend_time: " + endTime.Format(time.RFC3339)
 
-		postData = postData + "start_time: " + startTime.Format(time.RFC3339)
-		postData = postData + "\nend_time: " + endTime.Format(time.RFC3339)
+			if testURL != "" {
```

Comment:
When supporting gRPC, will the testURL be blank? Or be a function name? I think blank because it will just be a ping test.

Reply:
Hmm, interesting point. Another thing to consider is, maybe we might have to restructure the SMPS PerformanceTestConfig a little. The plan is to support 3 types of load test if I am correct: HTTP, TCP and gRPC. And, the three of them might have a different way to invoke the tests right, so a lot of the test configs must end up being specific to the test type? For example, request headers/cookie/body makes sense only in the case of HTTP and not in the other two! Similarly, we might have a specific config to TCP and gRPC too. I guess we should consider this in the SMPS!

@kushthedude any thoughts on this?

Reply:
Yes, we will likely need a traffic proto. For now, HTTP GET, gRPC ping, and TCP open/close can be supported by logic in `mesheryctl` only.